hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -917,6 +918,7 @@ public long getToVersion() {
     
     public static interface Validator {
         void validate(HollowProducer.ReadState readState);
+        void lock();
```

Comment:
I think a name like `lock()` should be reserved for concurrency type locks.

Let me discuss some naming ideas with @akhaku. My first thoughts are something like `freeze`, `freezeConfig`, `commitConfig`, `sealConfig`, …. Not real happy with any of them.